### DISCUSSION TITLE
Does it work with function based views?

### DISCUSSION THREAD

**Discussion by dchimeno** [jrief/django-formset#179] [Q&A] [answered]
After looking for it in documentation, all examples uses some kind of FormView.
is it possible to use function based views? Is there any example?

I'm looking to migrate an already existing project with hundreds of function based views...

**Comment by jrief** [comment, marked as answer]
As always, it depends.

If your are using a widget which requires an endpoint, such as the Selectize- or FileUpload-widget, then this extra endpoint must be handled somewhere. Therefore django-formset makes heavy use of special view mixin classes, such as [IncompleteSelectResponseMixin](https://github.com/jrief/django-formset/blob/b997b27c28e58609c5e777a2d66c1075247afcc8/formset/views.py#L23), [FileUploadMixin](https://github.com/jrief/django-formset/blob/b997b27c28e58609c5e777a2d66c1075247afcc8/formset/upload.py#L87).

For widgets which do not require an endpoint, some requests and responses must be altered. Here it might be possible to proceed with function based views. However, I believe it is less work to migrate them to classed based views rather than altering the request and response in those functions.
